api/filter_dimension_options.go

```diff
@@ -517,23 +517,46 @@ func (api *FilterAPI) patchFilterBlueprintDimensionHandler(w http.ResponseWriter
 	}
 
 	// apply the patches to the filter blueprint dimension options
-	successfulPatches, newETag, err := api.patchFilterBlueprintDimension(ctx, filterBlueprintID, dimensionName, patches, logData, eTag)
+	var newETag interface{}
+	newETag, err = api.dataStore.RunTransaction(ctx, true, func(transactionCtx context.Context) (interface{}, error) {
+		var (
+			allOptions, options []string
+			tag                 = eTag
+		)
+
+		// apply patch operations sequentially, stop processing if one patch fails, and return a list of successful patches operations
+		for _, patch := range patches {
+			allOptions, err = getStringArrayFromInterface(patch.Value)
+			if err != nil {
+				return tag, err
+			}
+			options = RemoveDuplicateAndEmptyOptions(allOptions)
+
+			if patch.Op == dprequest.OpAdd.String() {
+				tag, err = api.addFilterBlueprintDimensionOptions(transactionCtx, filterBlueprintID, dimensionName, options, logData, tag)
```

Comment:
Small thing but as Go style guides suggest keeping variable names short and transaction variables are often referred to as `tx` maybe `txCtx` would be a suitable name for the context